src/disclosure-policy/validateAttributeBasedAccessControlPolicy.ts

```diff
@@ -0,0 +1,36 @@
+import type { AgentContext, DcqlQuery } from '@credo-ts/core'
+import { SdJwtVcService } from '@credo-ts/core'
+import { type DcqlCredential, type DcqlQuery as Query, runDcqlQuery } from 'dcql'
+
+export type AttributeBasedAccessControlPolicy = {
+  attribute_based_access_control: DcqlQuery
+}
+
+export const validateAttributeBasedAccessControlPolicy = (
+  agentContext: AgentContext,
+  attributeBasedAccessControlPolicy: AttributeBasedAccessControlPolicy,
+  authorizationAttestations: Array<string>
+) => {
+  const credentials: DcqlCredential[] = []
+  const sdJwtService = agentContext.dependencyManager.resolve(SdJwtVcService)
+  for (const authorizationAttestation of authorizationAttestations) {
+    const sdJwtCredential = sdJwtService.fromCompact(authorizationAttestation)
```

Comment:
Do you filter the attestations beforehand, so we always know this will be an Sd-JWT?